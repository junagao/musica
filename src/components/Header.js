import React from 'react';

import './Header.scss';

const Header = () => (
  <header>
    <h1>
      <a href="/">musica</a>
    </h1>
  </header>
);

export default Header;
