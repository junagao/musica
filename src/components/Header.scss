@import '../variables';

h1 {
  color: $color-h1;
  font-size: 4.5rem;
  font-weight: $font-weight-h1;
  @media (min-width: $mobileL) {
    font-size: 5rem;
  }
  @media (min-width: $tabletP) {
    font-size: 5.5rem;
  }
}
