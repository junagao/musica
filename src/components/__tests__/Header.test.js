import React from 'react';
import { shallow } from 'enzyme';

import Header from 'components/Header';

describe('Header', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one header element', () => {
    expect(wrapper.find('header').length).toEqual(1);
  });

  it('should render one h1 element with the header title of `musica`', () => {
    expect(wrapper.find('h1').length).toEqual(1);
    expect(wrapper.find('h1').text()).toEqual('musica');
  });

  it('should have a link to /', () => {
    expect(wrapper.find('a').prop('href')).toEqual('/');
  });
});
