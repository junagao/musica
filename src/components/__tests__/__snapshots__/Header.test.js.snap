// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<header>
  <h1>
    <a
      href="/"
    >
      musica
    </a>
  </h1>
</header>
`;
